DOL.WHD.Section14c.Test/RepositoryMocks/ActivityLogRepositoryMock.cs

```diff
@@ -0,0 +1,78 @@
+        {
+            _data = new List<APIActivityLogs>
+            {
+                new APIActivityLogs {Id=1, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e02", EIN = "12-1234567", Exception = "My Test Exception 1", Level = "Fatal", Message="This a test message", User="test@test.com", UserId="12345"},
+                new APIActivityLogs {Id=2, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e03", EIN = "12-2234567", Exception = "My Test Exception 2", Level = "Error", Message="This a test message", User="test2@test.com", UserId="22345"},
+                new APIActivityLogs {Id=3, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e04", EIN = "12-3234567", Exception = "My Test Exception 3", Level = "Warn", Message="This a test message", User="test3@test.com", UserId="32345"},
+                new APIActivityLogs {Id=4, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e05", EIN = "12-4234567", Exception = "My Test Exception 4", Level = "Info", Message="This a test message", User="test4@test.com", UserId="42345"},
+                new APIActivityLogs {Id=5, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e06", EIN = "12-5234567", Exception = "My Test Exception 5", Level = "Debug", Message="This a test message", User="test5@test.com", UserId="52345"},
+                new APIActivityLogs {Id=6, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e07", EIN = "12-6234567", Exception = "My Test Exception 6", Level = "Trace", Message="This a test message", User="test6@test.com", UserId="62345"},
+            };
+        }
+
+        public IQueryable<APIActivityLogs> GetAllLogs()
+        {
+            return _data.AsQueryable();
+        }
+
+        public async Task<APIActivityLogs> GetActivityLogByIDAsync(int id)
+        {
+            return _data.AsQueryable().SingleOrDefault(x => x.Id == id);
+        }
+
+        public LogDetails AddLog(LogDetails entity)
+        {
+            if (entity != null)
+            {
+                LogEventInfo eventInfo = new LogEventInfo();
+
+                eventInfo.Properties["EIN"] = string.IsNullOrEmpty(entity.EIN) ? string.Empty : entity.EIN;
+                eventInfo.LoggerName = "NLog";
+                if (string.IsNullOrEmpty(entity.Message))
+                {
+                    throw new ArgumentException("Message cannot be null or empty string", "Log Message");
+                }
+
+                eventInfo.Message = entity.Message;
+
+                if (!string.IsNullOrEmpty(entity.Exception))
+                {
+                    eventInfo.Exception = new Exception(entity.Exception);
+                    eventInfo.SetStackTrace((new System.Diagnostics.StackTrace(new Exception(entity.Exception), false)), 1);
+                }
+
+
+                eventInfo.Level = LogLevel.FromString(entity.Level);
+                eventInfo.Properties["UserId"] = entity.UserId;
+                eventInfo.Properties["UserName"] = entity.User;
+            }
+            return entity;
+        }
```

Comment:
Since the `ActivityLogsController` never actually calls this `AddLog` method, would it be okay to stub it as simply this:

```C#
public LogDetails AddLog(LogDetails entity)
{
    return entity;
}
```

Since it's just to conform to the interface, it shouldn't actually need to do anything.  Having this much logic in a mock class is kind of a code smell.  How do you test the tests?

Reply:
@mgwalker The AddLog controller for ActivityLogs was no longer needed. I removed it from API and it's on my todo list to remove from interface and test. I was doing a few things at the same time, it fall off my radar. It will be remove from my next check in.